### PR TITLE
add export image function

### PR DESCRIPTION
**Pull request by fakedestinyck**
#51

### CODE REVIEW

File: src/app/schedule-view/component.ts [hunk 1]
```diff
@@ -147,7 +148,29 @@ export class ScheduleViewComponent implements OnInit, OnDestroy {
...
+          console.error('oops, something went wrong!', error);
+        });
+  }
+
   public clear (): void {
     this.selectionService.clear();
   }
 }
+
+@Component({
```
Comment (Bad-Science) [inline src/app/schedule-view/component.ts]: What's this?

File: src/app/schedule-view/component.ts [hunk 2]
```diff
@@ -147,7 +148,29 @@ export class ScheduleViewComponent implements OnInit, OnDestroy {
     return this.schedules[this.scheduleIndex].statusText;
   }
 
+  public downloadImage () : void {
+      var node = document.getElementById('schedule-container');
```
Comment (Bad-Science) [inline src/app/schedule-view/component.ts]: The conventional way to get DOM elements in Angular is using the ViewChild API. see: https://angular.io/api/core/ViewChild

File: src/app/schedule-view/component.html [hunk 3]
```diff
@@ -8,6 +8,19 @@
...
       <button class="schedule-button" (click) = "clear()">Clear</button>
+      <div ngbDropdown class="d-inline-block">
+          <button class="schedule-button" id="dropdownBasic1" ngbDropdownToggle>
+            Export as ...
+          </button>
+          <div ngbDropdownMenu>
+              <button class="schedule-button dropdown-item" (click) = "downloadImage()">Image</button>
+          </div>
+      </div>
+      <!-- <button class="schedule-button" (click) = "showExport()">Export as ...</button> -->
```
Comment (Bad-Science) [inline src/app/schedule-view/component.html]: Please remove commented code

File: src/app/schedule-view/component.html [hunk 4]
```diff
@@ -8,6 +8,19 @@
         <span class="schedule-switch" (click) = "nextSchedule()">&#9654;</span>
       </div>
       <button class="schedule-button" (click) = "clear()">Clear</button>
+      <div ngbDropdown class="d-inline-block">
+          <button class="schedule-button" id="dropdownBasic1" ngbDropdownToggle>
+            Export as ...
```
Comment (Bad-Science) [inline src/app/schedule-view/component.html]: Can we make this say "share", and then make the drop down item say "download image"

**Review by Bad-Science — CHANGES_REQUESTED** [review]
Thanks! Requested changes. Please add a description to the PR as well :)

**Comment by fakedestinyck** [comment]
@Bad-Science  Updated

**Comment by Bad-Science** [comment]
@fakedestinyck Thanks so much! This looks great, I'll merge it shortly! Would you mind just posting a screenshot of the button, and one of the downloaded schedule images in this thread for reference?

Thanks again!

**Review by Bad-Science — APPROVED** [review]

**Comment by fakedestinyck** [comment]
**Here is the _button_:**
screen shot 2018-03-30 at 15 28 57

**And the _downloaded schedule image_:**
myschedules

**Pull request merged**